scrapd/core/apd.py

```diff
@@ -179,14 +179,42 @@ def parse_twitter_description(twitter_description):
     return sanitize_fatality_entity(d)
 
 
+def parse_detail_page_description(details_description):
+    """
+    Clean up a details page notes section.
+
+    The purpose of this function is to attempt to extract the sentences about
+    the crash with some level of fidelity, but does not always return
+    a perfectly parsed sentence as the html syntax varies widely.
+
+    :param str details_description: the paragraph after the Deceased information
+    :return: A paragraph containing the details of the fatality in sentence form.
+    :rtype: str
+    """
+    start_tag = details_description.find('<p>') + 3
+    end_tag = details_description.find('</p>', start_tag)
+    if not details_description[start_tag:end_tag].upper().isupper():
+        start_tag = details_description.find(r'<br \>') + 6
+    snippet = details_description[start_tag:end_tag]
+    if snippet[:4] == '<img':
+        start_tag = details_description.find(r'<br \>') + 6
+    squished = details_description[start_tag:end_tag].replace('\n', ' ')
```

Comment:
Lines 196 to 200 are a little confusing. It seems you are updating the `start_tag` index if there is no `details_description` within the boundaries, or if the following tag is an image. But is there is no `details_description` anyway we are going to the end of the function to simply return an empty string.

What do you thing of rewriting it like that:

```python
start_tag = details_description.find('<p>') + 3
end_tag = details_description.find('</p>', start_tag)
snippet = details_description[start_tag:end_tag]

# Return an empty string if there is no snippet.
if not snippet:
    return ''

# Update the snippet if the following tag is an image.
if snippet[:4] == '<img':
    start_tag = details_description.find(r'<br \>') + 6
    snippet = details_description[start_tag:end_tag]

# Remove the end of line characters.
squished = snippet.replace('\n', ' ')

# Look for the first capital letter.
[...]
```